js/ui/windowManager.js

```diff
@@ -2205,6 +2223,33 @@ var WindowManager = new Lang.Class({
             this._hideOtherWindows(actor, animateFade);
     },
 
+    _clipActor : function(origin, position, actor) {
+        switch (origin) {
+            case "right": {
+                let clipx = position - actor.x;
+                actor.set_clip(0, 0, clipx, actor.height);
+                break;
+            }
+            case "left": {
+                let clipx = actor.width - (actor.x - position);
```

Comment:
Nitpick: maybe it should continue using the CamelCase fashion here (`clipX`) and the rest of the switch body too.